tests/tt_metal/tt_metal/unit_tests_fast_dispatch/common/command_queue_fixture.hpp

```diff
@@ -63,3 +63,35 @@ class CommandQueueMultiDeviceFixture : public ::testing::Test {
     tt::ARCH arch_;
     size_t num_devices_;
 };
+
+class CommandQueuePCIDevicesFixture : public ::testing::Test {
+   protected:
+    void SetUp() override {
+        auto slow_dispatch = getenv("TT_METAL_SLOW_DISPATCH_MODE");
+        if (slow_dispatch) {
+            TT_THROW("This suite can only be run with fast dispatch or TT_METAL_SLOW_DISPATCH_MODE unset");
+            GTEST_SKIP();
+        }
+        arch_ = tt::get_arch_from_string(tt::test_utils::get_env_arch_name());
+
+        num_devices_ = tt::tt_metal::GetNumPCIeDevices();
+        if (num_devices_ < 2) {
+            GTEST_SKIP();
+        }
+
+        for (unsigned int id = 0; id < num_devices_; id++) {
+            auto* device = tt::tt_metal::CreateDevice(id);
```

Comment:
How do these ids map down to HW if there are also remote chips in the system?

Reply:
Yea, for now it's assumed that PCI chips are always the first chips, so chips (0 - num_pci_chips) are the ids. It's not great, but this is an assumption at the syseng level too